apps/taskman/models.py

```diff
@@ -58,12 +75,22 @@ class Task(NullStrFieldsMixin):
     )
     flaw = models.OneToOneField(Flaw, on_delete=models.CASCADE, related_name="task")
 
+    @property
+    def workflow_state(self):
+        for key, value in WORKFLOW_TO_JIRA_STATUS.items():
+
+            if value == (self.status, self.resolution):
+                return key
+        raise KeyError(
+            "Current combination of Jira status and resolution does not met any workflow state mapped."
+        )
```

Comment:
Conceptual note. I am not sure it will be always possible to clearly map the task to its workflow state only base on the Jira status:resolution. I can imaging eg. default workflow and Major Incident workflow being only different in the Major Incident status value but otherwise sharing the same Jira status:resolution values. Workflow state should be I think rather something what you get when you call `classify` on flaw-task combination (or `classification` if you want the stored one).